Sources/Private/Experimental/Animations/ShapeAnimation.swift

```diff
@@ -98,3 +113,28 @@ extension CAShapeLayer {
       }, context: context)
   }
 }
+
+extension Trim {
+  /// Checks whether or not the value for `trim.start` is greater
+  /// than the value for every `trim.end` at every keyframe.
+  ///  - When this is the case, the start/end values need to be flipped
+  ///    when applied to a `CAShapeLayer`, which requires that
+  ///    `strokeStart` be less than `strokeEnd`.
+  fileprivate func startValueIsAlwaysGreaterThanEndValue() -> Bool {
+    let keyframeTimes = Set(start.keyframes.map { $0.time } + end.keyframes.map { $0.time })
+
+    let startInterpolator = KeyframeInterpolator(keyframes: start.keyframes)
+    let endInterpolator = KeyframeInterpolator(keyframes: end.keyframes)
+
+    for keyframeTime in keyframeTimes {
+      let startAtTime = startInterpolator.value(frame: keyframeTime) as! Vector1D
+      let endAtTime = endInterpolator.value(frame: keyframeTime) as! Vector1D
```

Comment:
nit: include a comment on why this force cast is safe?

Reply:
oh whoops I meant to clean this up, ty for the reminder